src/inference/tests/unit/cpu_reserve_test.cpp

```diff
@@ -0,0 +1,473 @@
+        {8, 0, 8, HYPER_THREADING_PROC, 8, -1},    {9, 0, 9, HYPER_THREADING_PROC, 9, -1},
+        {10, 0, 10, HYPER_THREADING_PROC, 10, -1}, {11, 0, 11, HYPER_THREADING_PROC, 11, -1},
+        {12, 0, 12, HYPER_THREADING_PROC, 12, -1}, {13, 0, 13, HYPER_THREADING_PROC, 13, -1},
+        {14, 0, 14, HYPER_THREADING_PROC, 14, -1}, {15, 0, 15, HYPER_THREADING_PROC, 15, -1},
+        {16, 0, 16, HYPER_THREADING_PROC, 16, -1}, {17, 0, 17, HYPER_THREADING_PROC, 17, -1},
+        {18, 1, 18, HYPER_THREADING_PROC, 18, -1}, {19, 1, 19, HYPER_THREADING_PROC, 19, -1},
+        {20, 1, 20, HYPER_THREADING_PROC, 20, -1}, {21, 1, 21, HYPER_THREADING_PROC, 21, -1},
+        {22, 1, 22, HYPER_THREADING_PROC, 22, -1}, {23, 1, 23, HYPER_THREADING_PROC, 23, -1},
+        {24, 1, 24, HYPER_THREADING_PROC, 24, -1}, {25, 1, 25, HYPER_THREADING_PROC, 25, -1},
+        {26, 1, 26, HYPER_THREADING_PROC, 26, -1}, {27, 1, 27, HYPER_THREADING_PROC, 27, -1},
+        {28, 1, 28, HYPER_THREADING_PROC, 28, -1}, {29, 1, 29, HYPER_THREADING_PROC, 29, -1},
+        {30, 1, 30, HYPER_THREADING_PROC, 30, -1}, {31, 1, 31, HYPER_THREADING_PROC, 31, -1},
+        {32, 1, 32, HYPER_THREADING_PROC, 32, -1}, {33, 1, 33, HYPER_THREADING_PROC, 33, -1},
+        {34, 1, 34, HYPER_THREADING_PROC, 34, -1}, {35, 1, 35, HYPER_THREADING_PROC, 35, -1},
+        {36, 0, 36, MAIN_CORE_PROC, 36, -1},       {37, 0, 37, MAIN_CORE_PROC, 37, -1},
+        {38, 0, 38, MAIN_CORE_PROC, 38, -1},       {39, 0, 39, MAIN_CORE_PROC, 39, -1},
+        {40, 0, 40, MAIN_CORE_PROC, 40, -1},       {41, 0, 41, MAIN_CORE_PROC, 41, -1},
+        {42, 0, 42, MAIN_CORE_PROC, 42, -1},       {43, 0, 43, MAIN_CORE_PROC, 43, -1},
+        {44, 0, 44, MAIN_CORE_PROC, 44, -1},       {45, 0, 45, MAIN_CORE_PROC, 45, -1},
+        {46, 0, 46, MAIN_CORE_PROC, 46, -1},       {47, 0, 47, MAIN_CORE_PROC, 47, -1},
+        {48, 0, 48, MAIN_CORE_PROC, 48, -1},       {49, 0, 49, MAIN_CORE_PROC, 49, -1},
+        {50, 0, 50, MAIN_CORE_PROC, 50, -1},       {51, 0, 51, MAIN_CORE_PROC, 51, -1},
+        {52, 0, 52, MAIN_CORE_PROC, 52, -1},       {53, 0, 53, MAIN_CORE_PROC, 53, -1},
+        {54, 1, 54, MAIN_CORE_PROC, 54, -1},       {55, 1, 55, MAIN_CORE_PROC, 55, -1},
+        {56, 1, 56, MAIN_CORE_PROC, 56, -1},       {57, 1, 57, MAIN_CORE_PROC, 57, -1},
+        {58, 1, 58, MAIN_CORE_PROC, 58, -1},       {59, 1, 59, MAIN_CORE_PROC, 59, -1},
+        {60, 1, 60, MAIN_CORE_PROC, 60, -1},       {61, 1, 61, MAIN_CORE_PROC, 61, -1},
+        {62, 1, 62, MAIN_CORE_PROC, 62, -1},       {63, 1, 63, MAIN_CORE_PROC, 63, -1},
+        {64, 1, 64, MAIN_CORE_PROC, 64, -1},       {65, 1, 65, MAIN_CORE_PROC, 65, -1},
+        {66, 1, 66, MAIN_CORE_PROC, 66, -1},       {67, 1, 67, MAIN_CORE_PROC, 67, -1},
+        {68, 1, 68, MAIN_CORE_PROC, 68, -1},       {69, 1, 69, MAIN_CORE_PROC, 69, -1},
+        {70, 1, 70, MAIN_CORE_PROC, 70, -1},       {71, 1, 71, MAIN_CORE_PROC, 71, -1},
+    },
+    {{9, MAIN_CORE_PROC, 4}},
```

Comment:
please add test cases for `{{3, MAIN_CORE_PROC, 4}}` and `{{4, MAIN_CORE_PROC, 4}}`